Src/PCompiler/CompilerCore/TypeChecker/Analyzer.cs

```diff
@@ -161,7 +161,7 @@ private static void ApplyPropagations<T>(IEnumerable<Function> functions, params
 
         private static Scope BuildGlobalScope(ICompilerConfiguration config, PParser.ProgramContext[] programUnits)
         {
-            var globalScope = Scope.CreateGlobalScope(config.Handler);
+            var globalScope = Scope.CreateGlobalScope(config.Handler, config.LocationResolver, config.ProjectDependencies);
```

Comment:
why dont we pass the config itself instead of passing 3 different parameters?

Reply:
Good point!
Done via 6758a4af0c3e39ef87622405bb8c78c12dea7a76.